examples/rust/components/http-jsonify/src/lib.rs

```diff
@@ -0,0 +1,168 @@
+            "method": method,
+            "path": path,
+            "query_string": query,
+            "body": body_bytes,
+        });
+
+        // Convert the JSON object to bytes
+        let resp_bytes =
+            serde_json::to_vec(&response_json).expect("failed to serialize response body JSON");
+
+        // Build the outgoing response
+        let outgoing_response = OutgoingResponse::new(Fields::new());
+        outgoing_response
+            .send_body(resp_bytes.as_slice())
+            .expect("failed to send response body");
+
+        // Set the response to be used
+        ResponseOutparam::set(outparam, Ok(outgoing_response));
+    }
+}
+
+// NOTE: Since wit-bindgen makes `IncomingRequest` available to us as a local type,
+// we can add convenience functions to it
+impl IncomingRequest {
+    /// This is a convenience function that writes out the body of a IncomingRequest (from wasi:http)
+    /// into anything that supports [`std::io::Write`]
+    fn read_body(self) -> Result<Vec<u8>> {
+        // Read the body
+        let incoming_req_body = self
+            .consume()
+            .map_err(|()| anyhow!("failed to consume incoming request body"))?;
+        let incoming_req_body_stream = incoming_req_body
+            .stream()
+            .map_err(|()| anyhow!("failed to build stream for incoming request body"))?;
+        let mut buf = Vec::<u8>::with_capacity(MAX_READ_BYTES as usize);
+        loop {
+            match incoming_req_body_stream.read(MAX_READ_BYTES as u64) {
+                Ok(bytes) => buf.extend(bytes),
+                Err(StreamError::Closed) => {
+                    buf.shrink_to_fit();
+                    break;
+                }
+                Err(e) => bail!("failed to read bytes: {e}"),
+            }
+        }
+        IncomingBody::finish(incoming_req_body);
+        Ok(buf)
+    }
+}
+
+// NOTE: Since wit-bindgen makes `OutgoingBody` available to us as a local type,
+// we can add convenience functions to it
+impl OutgoingResponse {
+    /// This is a convenience function that writes out the body of a IncomingRequest (from wasi:http)
+    /// into anything that supports [`std::io::Read`]
+    fn send_body(&self, mut buf: impl Read) -> Result<()> {
+        let body = self.body().expect("failed to open outgoing response body");
+        let out = body
+            .write()
+            .expect("failed to start writing to outgoing response body");
+        let mut spool = Vec::<u8>::with_capacity(MAX_READ_BYTES as usize);
+        loop {
+            match buf.read(&mut spool) {
+                // If we read the max bytes then we're clear to read more
+                Ok(bytes_read) => {
+                    let _ = out.blocking_write_and_flush(spool.as_slice());
+                    spool.clear();
+                    // If we read *less* than the maximum amount of bytes, we're at the end
+                    if bytes_read < MAX_READ_BYTES as usize {
```

Comment:
Generally you're supposed to read until you get a `bytes_read` of `0` or an `EOF`

Reply:
We could also just simplify this function and have it take the whole body as a param instead of an `impl Read`

Reply:
Hey @thomastaylor312 which body to you mean for replacing the `impl Read`? Are you thinking instead of the `impl Read` it should take a `Vec<u8>` (`resp_bytes`)? 

We talked about this previously but that what is returned is wrapped in JSON (w/ method, path, etc) -- would you like to return the request body directly instead?